networking/src/Ntp/Client.hs

```diff
@@ -167,20 +172,27 @@ sendLoop cli addrs = do
             -- wait for responses and update status
             _ <- timeout respTimeout waitForResponses
             updateStatus cli
+            -- after @'updateStatus'@ @'ntpStatus'@ is guaranteed to be
+            -- different from @'NtpSyncPending'@, now we can wait until it was
+            -- changed back to @'NtpSyncPending'@ to force a request.
+            waitForRequest
```

Comment:
One more question, this time about the `withAsync` concurrency here. Why is it needed? Could this not be done sequentially by `sendPacking`ing, then `timeout respTimeout waitForResponses`, then `waitForRequest`?

Reply:
That's true, unless the ntp servers answer faster than haskell executes `waitForResponses` ;)